### ISSUE TITLE
Plaidbench train vgg16 - Abort trap 6

### ISSUE THREAD

**Issue by andyoneal** [plaidml/plaidml#58]
Getting an "Abort trap: 6" error every time I run the vgg16 benchmark with training.

clinfo here: https://github.com/plaidml/plaidml/issues/17#issuecomment-338268111

on latest plaidml/plaidml-keras (0.1.2 from pip) and plaidbench.

> Andys-iMac-2:plaidbench andy$ python plaidbench.py --plaid --batch-size 16 --train vgg16
> Using PlaidML backend.
> 
> Current network being run : vgg16
> Loading the data
> INFO:plaidml:Opening device "amd_radeon_pro_580_compute_engine.0"
> Model loaded.
> Compiling and running initial batch, batch_size=16
> INFO:plaidml:Analyzing Ops: 57 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 108 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 204 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 217 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 230 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 242 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 249 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 262 of 483 operations complete
> Abort trap: 6

**Comment by andyoneal** [comment]
Hmm.. worked if I did batch size 8. Is failing with 16 expected then? A little strange that it ended with just 37% accuracy though.

> Andys-iMac-2:plaidbench andy$ python plaidbench.py --plaid --batch-size 8 --train vgg16
> Using PlaidML backend.
> 
> Current network being run : vgg16
> Loading the data
> INFO:plaidml:Opening device "amd_radeon_pro_580_compute_engine.0"
> Model loaded.
> Compiling and running initial batch, batch_size=8
> INFO:plaidml:Analyzing Ops: 57 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 125 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 211 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 223 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 234 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 242 of 483 operations complete
> INFO:plaidml:Analyzing Ops: 255 of 483 operations complete
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 5.5284 - acc: 0.0234         
> Doing the main timing
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 3.8217 - acc: 0.1172     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 2.8464 - acc: 0.2109     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 2.3874 - acc: 0.2266     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 2.1160 - acc: 0.2891     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 1.9297 - acc: 0.3203     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 1.7895 - acc: 0.3594     
> Epoch 1/1
> 128/128 [==============================] - 27s - loss: 1.6740 - acc: 0.3750     
> Example finished, elapsed: 30.6029119492 (compile), 218.338080168 (execution)

**Comment by jbruestle** [comment]
I'm fairly sure that what's causing the abort is out-of-memory.  Right now, our handling of  GPU memory limits isn't very good.  We have a new refactor coming soon that has a much more principled way to handle memory, which should prevent us from every requesting more memory than is available.

Regarding the accuracy, the plaidbench training code is really just to measure training performance, it's training on a very stupid dataset and no thought has been given to proper learning rates/etc.

I'll leave this ticket open until we land the memory fixes, after which point the message should be much more reasonable.

**Comment by brianretford** [comment]
Could you give our latest RC a try? We have done a bunch of work here: `pip install plaidml-keras==0.3.0rc1 plaidml==0.3.0rc1`

**Comment by tzerrell** [comment]
Closing this as stale; feel free to re-open if this issue persists
